internal/controllers/clusterextension_controller.go

```diff
@@ -306,12 +323,29 @@ func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alp
 		},
 	}
 
-	rel, state, err := r.getReleaseState(ac, ext, chrt, values, post)
+	rel, desiredRel, state, err := r.getReleaseState(ac, ext, chrt, values, post)
 	if err != nil {
 		setInstalledStatusConditionFailed(ext, fmt.Sprintf("%s:%v", ocv1alpha1.ReasonErrorGettingReleaseState, err))
 		return ctrl.Result{}, err
 	}
 
+	for _, preflight := range r.Preflights {
```

Comment:
Looks like we still need to add logic here to skip a preflight check if it has been disabled in the ClusterExtension spec configuration

Reply:
Are you referring to a top level check to not run all of the preflight checks if the check has been disabled in the ClusterExtension spec configuration? 

As of now, the way we have configured it in the ClusterExtension spec, we can disable all or none, and I don't think it's possible to skip individual preflight checks selectively, atleast based on the existing field we have in the API. Please correct me if I am wrong though.

Reply:
We added to the `ClusterExtension` API a method to disable the CRD Upgrade Safety preflight check specifically by creating a `ClusterExtension` that looks something like:
```yaml
apiVersion: olm.operatorframework.io/v1alpha1
kind: ClusterExtension
metadata:
  name: clusterextension-sample
spec:
  installNamespace: default
  packageName: argocd-operator
  version: 0.6.0
  preflight:
    crdUpgradeSafety:
      disabled: true
```

If we identify that the `ClusterExtension` is configured in this way we should ensure we are not running the preflight check if it is the CRD Upgrade Safety preflight check.

I am thinking we could maybe do a check like:
```go
...
if _, ok := preflight.(*crdupgradesafety.Preflight); ok && ext.Spec.Preflight.CRDUpgradeSafety.Disabled {
  // Skip this preflight check because it is of type *crdupgradesafety.Preflight and the CRD Upgrade Safety
  // preflight check has been disabled
  continue
}
...
```
at the top of the for loop

Reply:
Alternatively to that if statement, a switch statement could be utilized and is likely more easily extended than repeating if statements for each preflight check that can be disabled

Reply:
To clarify, when I say preflight check in this context I'm meaning the top level CRD Upgrade Safety preflight check and am not referring to the individual validations that are executed when running the CRD Upgrade Safety preflight check